### PR TITLE
BDD100K : Segmentation

### PR DESCRIPTION
**Pull request by ananno**
## Motivation and Context

Adding segmentation dataset and config file for bdd100k dataset

## Description

BDD100K dataset was used in object detection in blueoil but it also comes with segmentation data. To use bdd100k segmentation dataset a corresponding dataset class and a config file is implemented.

## How has this been tested?

`tox -e pytest`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ananno** [comment]
@iizukak San:
I began to separate #399 into multiple smaller PRs. This is the 3rd one. Please review.

**Comment by ananno** [comment]
@tkng, @iizukak San:
I've applied isort as requested. Please review one more time.

File: lmnet/lmnet/datasets/bdd100k.py [hunk 1]
```diff
@@ -155,13 +155,140 @@ def __len__(self):
...
+    @functools.lru_cache(maxsize=None)
+    def files_and_annotations(self):
+        split = "train"
+        if self.subset == "validation":
+            split = "val"
+        elif self.subset == "test":
+            split = "test"
+        file_path = os.path.join(self.data_dir, self.image_dir, split, "*.jpg")
+        image_paths = glob.glob(file_path)
+        image_paths.sort()
```
Comment (tkng) [inline lmnet/lmnet/datasets/bdd100k.py]: How about inserting an assert that checks the length of eact path list, like `assert(len(image_paths) == len(label_paths))`?

File: lmnet/lmnet/datasets/bdd100k.py [hunk 2]
```diff
@@ -155,13 +155,140 @@ def __len__(self):
...
+        return np.array([
+            unlabeled, dynamic, ego_vehicle, ground, static, parking, rail_track, road, sidewalk, bridge,
+            building, fence, garage, guard_rail, tunnel, wall, banner, billboard, lane_divider, parking_sign,
+            pole, polegroup, street_light, traffic_cone, traffic_device, traffic_light, traffic_sign,
+            traffic_sign_frame, terrain, vegetation, sky, person, rider, bicycle, bus, car, caravan,
+            motorcycle, trailer, train, truck])
+
+    @functools.lru_cache(maxsize=None)
+    def files_and_annotations(self):
+        split = "train"
```
[the remark this answers is not in the document]
Comment (ananno) [reply lmnet/lmnet/datasets/bdd100k.py]: Agreed. To be more specific I'm using `subset_dir` as variable name.

**Comment by ananno** [comment]
@iizukak, @tkng San, Please review one more time.

File: lmnet/lmnet/datasets/bdd100k.py (hunk 1 above)
Comment (tkng) [reply lmnet/lmnet/datasets/bdd100k.py]: @ananno How about this comment?

Comment (ananno) [reply lmnet/lmnet/datasets/bdd100k.py]: @tkng San: Sorry, but I've already added assert in line 280 as you suggested.

Comment (tkng) [reply lmnet/lmnet/datasets/bdd100k.py]: sorry, I missed.

**Pull request merged**